app/models/aleph_item.rb

```diff
@@ -72,3 +83,407 @@ def xml_status(id)
     Nokogiri::XML(open(status_url(id)))
   end
 end
+
+=begin
+A note about refactoring:
+
+The AlephItem class should actually be two classes:
+  * AlephResponse
+    * Processes XML from the Aleph API
+    * May represent one or more actual library objects
+    * Has a function which returns an array of AlephObjects in lieu of the
+      current `items` function
+  * AlephObject
+    * Is instantiated by AlephResponse with data extracted from XML
+    * Represents exactly one library object
+    * Contains all the methods that currently take `item` as an argument -
+      those actually want to be methods of an AlephItem class and in many cases
+      want to be setting or using instance variables (e.g. `@title`)
+
+Originally this class was only needed for AlephResponse-type things; it grew
+a lot of AlephObject-type things to support the needs of the full record view.
+
+We discussed refactoring it at this time but chose not to because:
+1) We believe we are nearing the end of our need to consume data from Aleph,
+   so the awkward developer UI will not slow us down as much as the refactor
+   would;
+2) We may be moving to a new ILS in the near-ish term, which would render a
+   refactor moot.
+
+If we find ourselves needing to do substantial new work with this class after
+the close of DI-513, we may wish to reconsider the decision not to refactor.
+
+If we do refactor it will have implications for the following:
+* AlephItem tests
+  * Will need to be renamed and separated out into AlephResponse and
+    AlephObject tests
+* AlephHelper
+  * Some of its responsibiilities may more properly be methods on AlephObject
+* views/aleph/full_item_status.html.erb
+  * This file expects items to be key/value hashes; it will need to expect
+    instead to get AlephObjects and access their instance variables or methods
+=end
+
+# Given an item, the ButtonMaker does the following:
+# * checks its eligibility for all actions
+# * when eligible, creates the HTML for a button powering that action
+# all_buttons returns the list of buttons thus created.
+# This means that we don't have to do any complex logic in the HTML - we can
+# just iterate through all available buttons and display them.
+# Important features:
+# @options is a list of the names of each action that might be available for an
+# item.
+# For each action in options, the following functions must exist:
+# * eligible_for_#{action}? - returns true/false
+# * make_button_for_#{action} - returns HTML of relevant button
+# This should probably have related to the Great Not-Completed AlephObject
+# Refactor, but it just got too janky to deal with all the edge cases in our
+# circ logic - too hard to tell whether they'd all been dealt with, except by
+# pulling it all into a special-purpose, encapsulated place.
+class ButtonMaker
```

Comment:
We should probably put the ButtonMaker class in it's own file. It's pretty big to just ride along in the AlephItem file.

Reply:
Done.